### PR TITLE
Feature/recipe test

### PR DESCRIPTION
**Pull request by lauraguerra1**
Description

- create tests and implementation code to find ingredient names for a given recipe 
- create tests and implementation code to find the cost of a given recipe 

Testing Notes

- each recipe card will load the ingredient names needed for the recipe and `getIngredients()` will be invoked to create recipe cards 
- each recipe card will display the total cost of the recipe and `caluclateRecipeCost()` will be invoked to create recipe cards 

Extra Notes 

- a helper function `getIngredientProperty()` was created to reuse logic across functions that need to match a given recipe's ingredients with the ingredient data to find a property (`name`, `estimatedCostInCents`) 
- sad path tests include checking the cost of a recipe with free ingredients
- sad path test was added to a previous iteration to check if a recipe with no instructions will return an empty array

### CODE REVIEW

**Comment by lauraguerra1** [comment]
[[Iteration1] Find ingredient names and cost per recipe](https://trello.com/c/V6E9iFx7/3-iteration1-find-ingredient-names-and-cost-per-recipe)

File: src/helper-functions.js [hunk 1]
```diff
@@ -1,3 +1,8 @@
 const copyItem = item => JSON.parse(JSON.stringify(item))
 
-export {copyItem}
\ No newline at end of file
+const getIngredientProperty = (currentIngredient, allIngredients, property) => {
+  let matchingIngredient = allIngredients.find(ingredient => ingredient.id === currentIngredient.id)
+  return matchingIngredient[property]
+}
```
Comment (mbenfowler) [inline src/helper-functions.js:6]: this is hot shit - we should probably test it too?

File: test/recipe-test.js [hunk 2]
```diff
@@ -1,13 +1,114 @@
...
+import {
+  getInstructions,
+  getIngredients,
+  calculateRecipeCost,
+} from '../src/recipes';
 
 describe('recipe', () => {
+  const cookies = sampleRecipeData[0];
+  const porkChops = sampleRecipeData[1];
+  const allIngredients = sampleIngredientsData;
```
Comment (mbenfowler) [inline test/recipe-test.js:19]: do we need to assign this?  Can we not just use the data import itself?

File: test/recipe-test.js [hunk 3]
```diff
@@ -1,13 +1,114 @@
...
+      {
+        id: 2,
+        estimatedCostInCents: 200,
+      },
+    ];
+    const cost = calculateRecipeCost(simpleRecipe, simpleIngredients)
+    assert.equal(cost, '$9.00')
+  });
+
+  it('should calculate the cost of a free recipe', () => {
```
Comment (mbenfowler) [inline test/recipe-test.js:79]: niiiice hahaha FREE FOOD!

File: test/recipe-test.js [hunk 4]
```diff
@@ -16,12 +117,20 @@ describe('recipe', () => {
...
+  it('should return instructions for a different recipe', () => {
+    const porkChopInstructions = getInstructions(porkChops);
     const porkChopDirections = [
       'Season the pork chops with salt and pepper and grill or pan fry over medium high heat until cooked, about 3-5 minutes per side. (If grilling, baste the chops in the maple dijon apple cider sauce as you grill.)Meanwhile, mix the remaining ingredients except the apple slices, bring to a simmer and cook until the sauce thickens, about 2-5 minutes.Grill or saute the apple slices until just tender but still crisp.Toss the pork chops and apple slices in the maple dijon apple cider sauce and enjoy!',
     ];
 
-    assert.deepEqual(cookieInstructions, cookieDirections);
     assert.deepEqual(porkChopInstructions, porkChopDirections);
   });
+  it('should return an empty array for recipes with no instructions', () => {
```
Comment (mbenfowler) [inline test/recipe-test.js:116]: good edge case

**Review by mbenfowler — CHANGES_REQUESTED** [review]

File: test/recipe-test.js (hunk 2 above)
Comment (taranveersingh93) [reply test/recipe-test.js:19]: @mbenfowler I found that assigning nested data to variables (with self-explanatory names) definitely helps with the readability especially if imports are involved.

**Review by taranveersingh93 — APPROVED** [review]
Love the dynamic nature of the functions and how readable they are even with complex under-the-hood tasks.

File: test/recipe-test.js (hunk 2 above)
Comment (lauraguerra1) [reply test/recipe-test.js:19]: What is the team consensus? Should I leave it as is or would you prefer me to use the imported data directly?

File: src/helper-functions.js (hunk 1 above)
Comment (lauraguerra1) [reply src/helper-functions.js:6]: good point! done now.

File: test/recipe-test.js (hunk 2 above)
Comment (mbenfowler) [reply test/recipe-test.js:19]: not a huge deal!  Whatever feels good to you :)

**Comment by lauraguerra1** [comment]
Additional Changes 

- A helper test file has been implemented to test the getIngredientProperty function.
- Extra data created (simpleRecipe and simpleIngredients) has been moved to the sample data file to be used across test suites

File: test/recipe-test.js (hunk 2 above)
Comment (lauraguerra1) [reply test/recipe-test.js:19]: I'll leave it this way just because it is easy to read

File: src/data/sampleData.js [hunk 5]
```diff
@@ -1107,3 +1107,38 @@ export const sampleUsersData = [
...
+  },
+];
+
+export {
+  sampleRecipeData,
+  sampleIngredientsData,
+  sampleUsersData,
+  simpleIngredients,
+  simpleRecipe
+}
```
Comment (mbenfowler) [inline src/data/sampleData.js]: newline!

**Review by mbenfowler — CHANGES_REQUESTED** [review]
just a single newline to add and looks great from me!

File: src/data/sampleData.js (hunk 5 above)
Comment (lauraguerra1) [reply src/data/sampleData.js]: Awesome! Done now.

**Review by mbenfowler — APPROVED** [review]
